### PR TITLE
Migrate availability schedule for everyone

### PR DESCRIPTION
**Pull request by KATT**
- Adds ~script~ **migration** that creates `Availability` for all users that are missing it
- The test can be deleted after it's been run
- Follow-ups: 
  - ~https://linear.app/cal/issue/CAL-699/availabilitystarttime-endtime-should-be-relative-to-users-time~
  - https://linear.app/cal/issue/CAL-700/remove-userstarttime-userendtime

### CODE REVIEW

File: pages/api/availability/__tmp__migration.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    // convert startTime/endTime to timezone using `user.timezone`
+    const baseDate = dayjs()
+      .tz(user.timeZone ?? "Europe/London")
+      .set("hour", 0)
+      .set("minute", 0)
+      .set("second", 0)
+      .set("millisecond", 0);
+
+    const startTime = baseDate.add(user.startTime, "minute").toDate();
+    const endTime = baseDate.add(user.endTime, "minute").toDate();
```
Comment (KATT) [inline pages/api/availability/__tmp__migration.ts:38]: cc @emrysal

File: package.json [hunk 2]
```diff
@@ -102,7 +102,7 @@
     "@types/accept-language-parser": "1.5.2",
     "@types/async": "^3.2.9",
     "@types/bcryptjs": "^2.4.2",
-    "@types/jest": "^27.0.1",
+    "@types/jest": "^26.0.0",
```
Comment (KATT) [inline package.json:105]: Downgraded b/c of https://github.com/prisma/prisma/issues/8558

File: prisma/migrations/20211115182559_availability_issue/migration.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  (
+    select 
+      users.id, 
+      users."startTime", 
+      users."endTime", 
+      users."timeZone",
+      count("Availability".id) as availability_count
+    from users 
+    left join "Availability" on "Availability"."userId" = users.id
+    where users."timeZone" != ''
```
Comment (KATT) [inline prisma/migrations/20211115182559_availability_issue/migration.sql:18]: - There might be a hidden issue here as dayjs' timezones is not guaranteed to match Postgres built-in timezones & we don't _actually_ validate that the input is a timezone
- I'd ideally do something like `WHERE users.timeZone IS VALID TIMEZONE` here but I can't google my way on how to do that here

File: prisma/migrations/20211115182559_availability_issue/migration.sql [hunk 4]
```diff
@@ -0,0 +1,21 @@
+
+insert into "Availability" ("userId", "startTime", "endTime", "days")
+select
+  id as "userId", 
+  CAST(CONCAT(CAST(("startTime" / 60) AS text), ':00') AS time) AT TIME ZONE "timeZone" AT TIME ZONE 'UTC',
+  CAST(CONCAT(CAST(("endTime" / 60) AS text), ':00') AS time) AT TIME ZONE "timeZone" AT TIME ZONE 'UTC',
```
Comment (KATT) [inline prisma/migrations/20211115182559_availability_issue/migration.sql:6]: 1. takes `startTime` (minutes from midnight)
2. converts it to a `TIME`
3. converts it from timezone `user.timeZone` --> `UTC`

**Comment by PeerRich** [comment]
whats the status of this?

**Comment by KATT** [comment]
> whats the status of this?

This PR doesn't actually fix it for everyone but rather migrates from old User start/end time to new model. 

@emrysal said he'd get the proper fixes out today, I think he was gonna merge this into his branch

File: prisma/migrations/20211115182559_availability_issue/migration.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      users.id, 
+      users."startTime", 
+      users."endTime", 
+      users."timeZone",
+      count("Availability".id) as availability_count
+    from users 
+    left join "Availability" on "Availability"."userId" = users.id
+    group by users.id
+  ) usersWithAvailabilityNumber
+where availability_count < 1
```
Comment (KATT) [inline prisma/migrations/20211115182559_availability_issue/migration.sql:20]: Excludes everyone that already has any `Availability`

File: prisma/migrations/20211115182559_availability_issue/migration.sql [hunk 6]
```diff
@@ -0,0 +1,20 @@
+
+insert into "Availability" ("userId", "startTime", "endTime", "days")
+select
+  id as "userId", 
+  CAST(CONCAT(CAST(("startTime") AS text), ' minute')::interval AS time) as "startTime",
+  CAST(CONCAT(CAST(("endTime") AS text), ' minute')::interval AS time) as "endTime",
```
Comment (KATT) [inline prisma/migrations/20211115182559_availability_issue/migration.sql:6]: converts i.e.
- `0` to `00:00:00` 
- `1439` to `23:59:00`

**Review by KATT — COMMENTED** [review]
code review hints

**Review by zomars — APPROVED** [review]
Looking good on my side

**Comment by PeerRich** [comment]
this is obsolete, right @KATT ?

**Comment by KATT** [comment]
> this is obsolete, right @KATT ?

Not really, it will helps us delete those legacy fields on Prisma.

I've wrote in the standups two weeks in a row that I want to merge this and waiting for more reviews.

**Pull request merged**